app/src/main/java/com/example/habitsmasher/DaysTracker.java

```diff
@@ -56,11 +57,29 @@ public DaysTracker(String days){
         setFalse();
 
         //set the specific days
-        setDays(days);
+        setDaysFromString(days);
 
 
     }
 
+    /**
+     * Creates a DaysTracker with arrayList input. This version of the constructor is used to convert data
+     * from the database to an object.
+     * @param days The list of days as {@link String}
+     */
+    public DaysTracker(ArrayList<String> days){
+
+
+        //create the arrayList
+        _days = new ArrayList<Boolean>(Arrays.asList(new Boolean[7]));
+
+        //set all to false
+        setFalse();
+
+        // convert from string to bool
+        setDaysFromList(days);
+    }
```

Comment:
There's a new way of setting the days of the week: ArrayList of strings. This is needed to conform to the way we populate the lists of habits (Through the queries to the DB). Now we just query based on whether the current day shows up in the array of days in the DB. I'm keeping the string based constructors and the like, however.